gems/sorbet-runtime/lib/types/private/abstract/declare.rb

```diff
@@ -31,15 +31,16 @@ def self.declare_abstract(mod, type:)
         raise "You must call `abstract!` *before* defining an initialize method"
       end
 
-      # Don't need to silence warnings via without_ruby_warnings when calling
-      # define_method because of the guard above
-
-      mod.send(:define_method, :initialize) do |*args, &blk|
-        if self.class == mod
-          raise "#{mod} is declared as abstract; it cannot be instantiated"
+      initialize_check_module = Module.new do
+        define_method(:initialize) do |*args, &blk|
+          if mod.equal?(self.class)
```

Comment:
I could be rigorous by doing `Object.instance_method(:class).bind_call(self)` here, but we don't have bind_call yet and I don't want to allocate.